lib/modules/datasource/go/releases-goproxy.ts

```diff
@@ -198,13 +204,18 @@ export class GoProxyDatasource extends Datasource {
     return input.replace(regEx(/([A-Z])/g), (x) => `!${x.toLowerCase()}`);
   }
 
-  async listVersions(baseUrl: string, packageName: string): Promise<string[]> {
+  async listVersions(baseUrl: string, packageName: string): Promise<Release[]> {
     const url = `${baseUrl}/${this.encodeCase(packageName)}/@v/list`;
     const { body } = await this.http.get(url);
     return body
-      .split(regEx(/\s+/))
-      .filter(Boolean)
-      .filter((x) => x.indexOf('+') === -1);
+      .split(newlineRegex)
+      .filter(is.nonEmptyStringAndNotWhitespace)
+      .map((str) => {
+        const [version, releaseTimestamp] = str.split(regEx(/\s+/));
+        return DateTime.fromISO(releaseTimestamp).toISO()
```

Comment:
```suggestion
        return releaseTimestamp && DateTime.fromISO(releaseTimestamp).isValid()
```

i think you like to validate here 😉

Reply:
Yeah, at first I was returning `fromISO().toISO()` result instead of original value